File: my-folder/0349-intersection-of-two-arrays/solution.py
class Solution(object):
    def intersection(self, nums1, nums2):
        a=[]
        for i in range(max(len(nums1),len(nums2))):
            if len(nums1)>len(nums2):
                if nums1[i] in nums2:
                    a.append(nums1[i])
            else:
                if nums2[i] in nums1:
                    a.append(nums2[i])
        return list(set(a))

File: my-folder/0349-intersection-of-two-arrays/solution.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        List<Integer> newArray = new ArrayList<>();
        Arrays.sort(nums1); 
        Arrays.sort(nums2);
        int i = 0, j = 0;
        int n1 = nums1.length, n2 = nums2.length;

        while (i < n1 && j < n2) {
            if(nums1[i]<nums2[j]){
                i++;
            }

            else if(nums1[i]>nums2[j]){
                j++;
            }
            
            else{
                if(newArray.size()==0 || newArray.get(newArray.size()-1)!=nums1[i]){
                    newArray.add(nums1[i]);
                }
                i++;
                j++;
            }
            
        }


        // Convert List to int[]
        int[] result = new int[newArray.size()];
        for (int k = 0; k < newArray.size(); k++) {
            result[k] = newArray.get(k);
        }

 
        return result;
    }
}

